ambari-server/src/main/java/org/apache/ambari/server/scheduler/ExecutionScheduleManager.java

```diff
@@ -275,13 +290,42 @@ public boolean continueOnMisfire(JobExecutionContext jobExecutionContext) {
     return true;
   }
 
+
+  private long getFirstJobOrderId(RequestExecution requestExecution) throws AmbariException {
+    Long firstBatchOrderId = null;
+    Batch batch = requestExecution.getBatch();
+    if (batch != null) {
+      List<BatchRequest> batchRequests = batch.getBatchRequests();
+      if (batchRequests != null) {
+        Collections.sort(batchRequests);
+        ListIterator<BatchRequest> iterator = batchRequests.listIterator();
+        firstBatchOrderId = iterator.next().getOrderId();
+      }
+    }
+    if (firstBatchOrderId == null) {
+      throw new AmbariException("Can't schedule RequestExecution with no batches");
```

Comment:
Can you add comment how this could ever happen?

Reply:
If the user creates a request schedule with no batches, this shouldn't ever happen under the normal circumstances, the check is just to make sure we deal with the situation